test/functional/eval/null_spec.lua

```diff
@@ -61,7 +61,7 @@ describe('NULL', function()
     null_expr_test('does not crash append()', 'append(1, L)', 0, 0, function()
       eq({''}, curbufmeths.get_lines(0, -1, false))
     end)
-    null_expr_test('does not crash setline()', 'setline(1, L)', 0, 0, function()
+    null_expr_test('does not crash setline()', 'setline(1, L)', 0, 1, function()
```

Comment:
It changes the behavior of `:echo setline(1, v:_null_list)` - which is fine I guess, and rather constructed, isn't it?

Reply:
changes it from what to what? Ideally, a null list should behave like a (locked) empty list.

Reply:
From 0 to 1 - i.e. it returns "error" instead of "ok".

> Ideally, a null list should behave like a (locked) empty list.

Ok, but that's behavior different from Vim then.
Due to the check in https://github.com/neovim/neovim/pull/10250/files#diff-deb695d507653839019814f54242d47bR14878.

Reply:
Previously it was using `if (argvars[1].v_type == VAR_LIST) {`.

Happy to keep it (and forward it to Vim), but it's the first I hear about null-lists.. :)

Reply:
> Ok, but that's behavior different from Vim then.

We shouldn't regress from improvements made in nvim. Vim displays "null" list as if they were empty lists, which is their intentional behavior. The only reason one needs to "hear" about null list is because it behaves differently in some edge cases for no principal reason.

Reply:
> Vim displays "null" list as if they were empty lists

But here it will cause a failure, i.e. it is not handled as if it is an empty list at least.
So it could be considered a bug there then.

As for the behavior in nvim it looks more like a side effect of not crashing on them.
check ceb45a088 / f572bd7e4e. (https://github.com/neovim/neovim/pull/10250/commits/b7584360705ef53a1555a2b7b5410c8339f9e08c)

Reply:
> As for the behavior in nvim it looks more like a side effect of not crashing on them.

What does this even mean? In what way is accepting the list more "a side effect of not crashing" than throwing an error would be? How would this motivate one (non-crashing) behavior over another?

Reply:
From @ZyX-I's commits it looked like he first refactored it, and when it crashed fixed that - but not using the condition that Vim had originally.

Reply:
Sure, but the general strategy he followed was to make null list behave consistently like locked empty lists, rather than just pick any non-crashing behavior arbitrarily.

Reply:
Ok, reverted.